src/main/java/com/avispl/dal/communicator/polycom/videoos/PolycomVideoOS.java

```diff
@@ -0,0 +1,1055 @@
+                if(conferences != null && conferences.size() > 0){
+                    callStatus.setCallId(getJsonProperty(conferences.get(0), "id", String.class));
+                    callStatus.setCallStatusState(CallStatus.CallStatusState.Connected);
+                    return callStatus;
+                }
+            }
+        } finally {
+            controlOperationsLock.unlock();
+        }
+        return callStatus;
+    }
+
+    @Override
+    public MuteStatus retrieveMuteStatus() throws Exception {
+        controlOperationsLock.lock();
+        try {
+            Boolean muted = doGet(buildHttpUrl(AUDIO_MUTED), Boolean.class);
+            if (muted) {
+                return MuteStatus.Muted;
+            } else {
+                return MuteStatus.Unmuted;
+            }
+        } finally {
+            controlOperationsLock.unlock();
+        }
+    }
+
+    @Override
+    public void sendMessage(PopupMessage popupMessage) throws Exception {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void mute() throws Exception {
+        controlOperationsLock.lock();
+        try {
+            doPost(AUDIO_MUTED, true);
+        } finally {
+            controlOperationsLock.unlock();
+        }
+    }
+
+    @Override
+    public void unmute() throws Exception {
+        controlOperationsLock.lock();
+        try {
+            doPost(AUDIO_MUTED, false);
+        } finally {
+            controlOperationsLock.unlock();
+        }
+    }
+
+    @Override
+    public List<Statistics> getMultipleStatistics() throws Exception {
+        ExtendedStatistics extendedStatistics = new ExtendedStatistics();
+        EndpointStatistics endpointStatistics = new EndpointStatistics();
+
+        controlOperationsLock.lock();
+        try {
+            if (isValidControlCoolDown() && localStatistics != null && localEndpointStatistics != null) {
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Device is occupied. Skipping statistics refresh call.");
+                }
+                extendedStatistics.setStatistics(localStatistics.getStatistics());
+                extendedStatistics.setControllableProperties(localStatistics.getControllableProperties());
+
+                endpointStatistics.setInCall(localEndpointStatistics.isInCall());
+                endpointStatistics.setCallStats(localEndpointStatistics.getCallStats());
+                endpointStatistics.setVideoChannelStats(localEndpointStatistics.getVideoChannelStats());
+                endpointStatistics.setAudioChannelStats(localEndpointStatistics.getAudioChannelStats());
+
+                return Arrays.asList(extendedStatistics, endpointStatistics);
+            }
+
+            Map<String, String> statistics = new HashMap<>();
+            List<AdvancedControllableProperty> controls = new ArrayList<>();
+            retrieveSystemStatus(statistics);

```

Comment:
How many calls you need to make during each monitoring cycle?!
How much time it will take?

Reply:
Current amount of calls takes ~5 seconds, with 160ms+ connection (guess my vpn slows it down a bit also)

Reply:
Hmm. It is not reasonable to hold calling thread for such amount of time. My personal rule of thumb (which can vary of course, also, not always can be achieved) is 1 sec max for synchronous calls. It looks like in this case this has do be done asynchronously. But I can live with it in v1 as long as it is documented (and it is not) - caller must be aware of that!

Reply:
Agree, done some additional checks + javadocs